### PR TITLE
Added method any(spec) 

### PR DESCRIPTION
**Pull request by islas27**
It  closes issue #3
- Added new method any(spec) to Array.prototype
- Added the unit test
- Added the documentation
- Added a note about running the examples on the documentation.

### CODE REVIEW

File: aee.js [hunk 1]
```diff
@@ -5,3 +5,17 @@ Array.prototype.each = function (callback) {
   }
   return this
 }
+
+Array.prototype.any = function (spec) {
+  const len = this.length
+  if (typeof spec !== 'function') {
+    searchedValue = spec
```
Comment (ryanez) [inline aee.js]: I can't find declaration for `searchValue`, is it a global variable?

File: aee.js [hunk 2]
```diff
@@ -5,3 +5,17 @@ Array.prototype.each = function (callback) {
...
+    spec = function (element, index) {
```
Comment (ryanez) [inline aee.js]: You can omit the parameters that you are not planning to use. I mean, while you don't change the position of the ones you need to use.

Comment (ryanez) [inline aee.js]: Do not modify parameters reference, use a new var instead, e.g.:
```js
let x = superCondition ? spec : value => value === spec;
```

File: examples/any.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+const people = [
+  { 'name': 'Pedro', 'age': 19 },
+  { 'name': 'Juan', 'age': 15 },
+  { 'name': 'Pablo', 'age': 16 },
+  { 'name': 'Pancho', 'age': 20 },
+  { 'name': 'Topo', 'age': 18 },
+  { 'name': 'Jonathan', 'age': 24 }
+]
+
+console.log(people.any((person, index) => person.name === 'Topo'))
```
Comment (ryanez) [inline examples/any.js]: `index` is declared but not used

File: examples/any.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+// Output:
+// true
+
+function spec (person, index) {
```
Comment (ryanez) [inline examples/any.js]: Same here.

File: test/any.spec.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+const expect = require('chai').expect
+const sinon = require('sinon')
+const data = require('./data')
+
+describe('#any', () => {
+  it('checks if any() is true when spec is a function and a match was found', () => {
+    function spec (element, index) {
+      return element.age === 24
+    }
+    var specSpy = sinon.spy(spec)
```
Comment (ryanez) [inline test/any.spec.js]: Or you can even use the arrow function as argument of the `sinon.spy`, e.g.:
```js
let spied = sinon.spy(element => element.age === 24)
```

File: test/any.spec.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+      return element.age === 80
+    }
+    var specSpy = sinon.spy(spec)
+    expect(data.people.any(specSpy)).to.equal(false)
+    expect(specSpy.callCount).to.equal(6)
+  })
+
+  it('checks if any() is true when spec is a value and it matched something', () => {
+    const spec = 'b'
+    expect(data.letters.any(spec)).to.equal(true)
```
Comment (ryanez) [inline test/any.spec.js]: you can use the constant literal `'b'` as `.any` argument, no one will result injured.
```js
expect(data.letters.any('b')).to.equal(true)
```

File: test/any.spec.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
+require('../aee.js')
+const expect = require('chai').expect
+const sinon = require('sinon')
+const data = require('./data')
+
+describe('#any', () => {
+  it('checks if any() is true when spec is a function and a match was found', () => {
+    function spec (element, index) {
```
Comment (ryanez) [inline test/any.spec.js]: Arrow functions are a beauty, e.g.:
```js
const spec = element => element.age === 24
```

File: test/any.spec.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+      return element.age === 24
+    }
+    var specSpy = sinon.spy(spec)
+    expect(data.people.any(specSpy)).to.equal(true)
+    expect(specSpy.callCount).to.equal(6)
```
Comment (ryanez) [inline test/any.spec.js]: You could try to be less egocentric and find someone else than yourself on the list of persons, that will allow you to test your code stops calling `spec` when it finds the match.

File: aee.js (hunk 1 above)
Comment (islas27) [reply aee.js]: fixed in commit c06539f

**Comment by islas27** [comment]
Thanks for the feedback @ryanez, already done with the fixes and added them in the commit c06539f.

**Comment by ryanez** [comment]
LGTM 👍

**Pull request merged**